internal/services/controller/controller_test.go

```diff
@@ -63,14 +61,127 @@ func TestMain(m *testing.M) {
 }
 
 func TestController_HappyPath(t *testing.T) {
-	scheme := runtime.NewScheme()
-	utilruntime.Must(karpenterCoreAlpha.SchemeBuilder.AddToScheme(scheme))
-	utilruntime.Must(karpenterAlpha.SchemeBuilder.AddToScheme(scheme))
-	utilruntime.Must(karpenterCore.SchemeBuilder.AddToScheme(scheme))
-	utilruntime.Must(karpenter.SchemeBuilder.AddToScheme(scheme))
-	utilruntime.Must(datadoghqv1alpha1.SchemeBuilder.AddToScheme(scheme))
-	utilruntime.Must(argorollouts.SchemeBuilder.AddToScheme(scheme))
+	tests := map[string]struct {
+		objectCount int
+	}{
+		"happy path": {
```

Comment:
minor: please use more explicit names for test cases.

`happy path` is too vague, don't give me a hint about what is conditions and meaning of happy path